BossMod/Modules/Endwalker/TreasureHunt/GymnasiouPithekos.cs

```diff
@@ -0,0 +1,126 @@
+using System.IO.Compression;
+using System.Security.Cryptography.X509Certificates;
+using FFXIVClientStructs.FFXIV.Common.Lua;
+
+namespace BossMod.Endwalker.TreasureHunt.GymnasiouPithekos
+{
+    public enum OID : uint
+    {
+        Boss = 0x3D2B, //R=6
+        BallOfLevin = 0x3E90,
+        BossAdd = 0x3D2C, //R=4.2
+        BossHelper = 0x233C,
+    };
+
+public enum AID : uint
+{
+    Attack = 872, // Boss->player, no cast, single-target
+    Thundercall = 32212, // Boss->location, 2,5s cast, range 3 circle
+    LightningBolt = 32214, // Boss->self, 3,0s cast, single-target
+    LightningBolt2 = 32215, // BossHelper->location, 3,0s cast, range 6 circle
+    ThunderIV = 32213, // BallOfLevin->self, 7,0s cast, range 18 circle
+    Spark = 32216, // Boss->self, 4,0s cast, range 14-30 donut --> TODO: confirm inner circle size
+    Attack2 = 870, // BossMikros->player, no cast, single-target
+    RockThrow = 32217, // BossMikros->location, 3,0s cast, range 6 circle
+    SweepingGouge = 32211, // Boss->player, 5,0s cast, single-target
+};
+
+public enum IconID : uint
+{    
+    AOE = 111, // Thundercall marker
+};
+
+    class Spark : Components.SelfTargetedAOEs
+    {
+        public Spark() : base(ActionID.MakeSpell(AID.Spark), new AOEShapeDonut(13,30)) { } 
+    }
+    class SweepingGouge : Components.SingleTargetCast
+    {
+        public SweepingGouge() : base(ActionID.MakeSpell(AID.SweepingGouge)) { }
+    }
+    class Thundercall : Components.LocationTargetedAOEs
+    {
+        private int targeted;
+        public Thundercall() : base(ActionID.MakeSpell(AID.Thundercall), 3) {}
+        public override void OnEventIcon(BossModule module, Actor actor, uint iconID)
+        {
+            var player = module.Raid.Player();
+            if(player == actor && iconID == (uint)IconID.AOE)
+                targeted = 1;
+        }
+        public override void OnCastStarted(BossModule module, Actor caster, ActorCastInfo spell)
+        {
+            if ((AID)spell.Action.ID == AID.Thundercall)
+                targeted = 0;
+        }
+        public override void AddGlobalHints(BossModule module, GlobalHints hints)
+        {
+             if (targeted > 0)
+                hints.Add("Drop the puddle at the edge of the arena and get away, it will spawn a huge AOE circle");
```

Comment:
These kind of things should be personal hints, rather than global (because it's only relevant to the spell target).
Also I tend to make hints very concise (eg. 'GTFO to the edge'), since you wouldn't really have time to read long full sentences mid fight

Reply:
well i made a target check (if (targeted > 0), thought that would be fine. but i can change it